lighthouse/constants.py

```diff
@@ -37,19 +36,35 @@
 MLWH_LH_SAMPLE_RNA_ID = "rna_id"
 MLWH_LH_SAMPLE_RESULT = "result"
 
+# Used for filtering positive results
 CT_VALUE_LIMIT = 30
 
-POSITIVE_SAMPLES_MONGODB_FILTER = {
-    FIELD_RESULT: {"$regex": "^positive", "$options": "i"},
-    FIELD_ROOT_SAMPLE_ID: {"$not": re.compile("^CBIQA_")},
-    "$or": [
-        {"$and": [{FIELD_CH1_CQ: None}, {FIELD_CH2_CQ: None}, {FIELD_CH3_CQ: None}]},
-        {
-            "$or": [
-                {FIELD_CH1_CQ: {"$lte": CT_VALUE_LIMIT}},
-                {FIELD_CH2_CQ: {"$lte": CT_VALUE_LIMIT}},
-                {FIELD_CH3_CQ: {"$lte": CT_VALUE_LIMIT}},
-            ]
-        },
-    ],
+
+STAGE_MATCH_POSITIVE = {
+    "$match": {
+        #  1. We are only interested in positive samples
+        FIELD_RESULT: {"$regex": "^positive", "$options": "i"},
+        # 2. We are not interested in controls
+        FIELD_ROOT_SAMPLE_ID: {"$not": {"$regex": "^CBIQA_"}},
+        # 3. Further filter the positive samples
+        # TODO: needs to align with the crawler changes
+        "$or": [
+            {
+                "$and": [
+                    {FIELD_CH1_CQ: {"$exists": False}},
+                    {FIELD_CH2_CQ: {"$exists": False}},
+                    {FIELD_CH3_CQ: {"$exists": False}},
+                ],
+            },
+            {
+                "$or": [
+                    {FIELD_CH1_CQ: {"$lte": CT_VALUE_LIMIT}},
+                    {FIELD_CH2_CQ: {"$lte": CT_VALUE_LIMIT}},
+                    {FIELD_CH3_CQ: {"$lte": CT_VALUE_LIMIT}},
+                ],
+            },
+        ],
+        # 4. We are only interested in documents which have a valid date
+        FIELD_DATE_TESTED: {"$exists": True, "$nin": [None, ""]},
```

Comment:
Just to let you know we also have the sample creation date  (Field ```created_at```) That is the date when the sample was first recorded in the Mongo database. What to use depends in what the SSRs are more interested, if they want the recent samples by testing date, or the recent samples by how we are receiving them (created_at).

Reply:
Rich (SSR) said they would like to use the "Date Tested" date.